components/connectivity-adapter/internal/appregistry/service/graphql.go

```diff
@@ -31,52 +36,84 @@ type gqlGetApplicationResponse struct {
+func (r *gqlRequester) SetApplicationLabel(appID string, label graphql.LabelInput) error {
+	gqlRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+			result: setApplicationLabel(applicationID: "%s", key: "%s", value: %s) {
+					%s
+				}
+			}`,
+			appID, label.Key, label.Value, r.gqlFieldsProvider.ForLabel()))
+
+	err := r.cli.Run(context.Background(), gqlRequest, nil)
 	if err != nil {
-		return nil, errors.Wrapf(err, "while constructing input")
+		return errors.Wrap(err, "while doing GraphQL request")
 	}
 
-	return gcli.NewRequest(
-		fmt.Sprintf(`mutation {
-			result: registerApplication(in: %s) {
-				id
-			}	
-		}`,
-			appInputGQL)), nil
+	return nil
 }
 
-func (b *gqlRequestBuilder) UnregisterApplicationRequest(id string) *gcli.Request {
-	return gcli.NewRequest(
+func (r *gqlRequester) CreateAPIDefinition(appID string, apiDefinitionInput graphql.APIDefinitionInput) (string, error) {
+	inStr, err := r.graphqlizer.APIDefinitionInputToGQL(apiDefinitionInput)
+	if err != nil {
+		return "", errors.Wrap(err, "while preparing GraphQL input")
+	}
+
+	gqlRequest := gcli.NewRequest(
 		fmt.Sprintf(`mutation {
-		result: unregisterApplication(id: "%s") {
-			id
-		}	
-	}`, id))
+			result: addAPIDefinition(applicationID: "%s", in: %s) {
+					%s
+				}
+			}`, appID, inStr, r.gqlFieldsProvider.ForAPIDefinition()))
+
+	var resp struct {
+		Result graphql.APIDefinition `json:"result"`
+	}
+
+	err = r.cli.Run(context.Background(), gqlRequest, &resp)
+	if err != nil {
+		return "", errors.Wrap(err, "while doing GraphQL request")
+	}
+
+	return resp.Result.ID, nil
 }
 
-func (b *gqlRequestBuilder) GetApplicationRequest(id string) *gcli.Request {
-	return gcli.NewRequest(
-		fmt.Sprintf(`query {
-			result: application(id: "%s") {
-					%s
-			}
-		}`, id, b.gqlFieldsProvider.ForApplication()))
+func (r *gqlRequester) CreateEventDefinition(appID string, eventDefinitionInput graphql.EventDefinitionInput) (string, error) {
+	inStr, err := r.graphqlizer.EventDefinitionInputToGQL(eventDefinitionInput)
+	if err != nil {
+		return "", errors.Wrap(err, "while preparing GraphQL input")
+	}
+
+	gqlRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: addEventDefinition(applicationID: "%s", in: %s) {
+						%s	
+					}
+				}`, appID, inStr, r.gqlFieldsProvider.ForEventDefinition()))
+
+	var resp struct {
+		Result graphql.EventDefinition `json:"result"`
+	}
+
+	err = r.cli.Run(context.Background(), gqlRequest, &resp)
+	if err != nil {
+		return "", errors.Wrap(err, "while doing GraphQL request")
+	}
+
+	return resp.Result.ID, nil
 }
 
-func (b *gqlRequestBuilder) GetApplicationsByName(appName string) *gcli.Request {
-	return gcli.NewRequest(fmt.Sprintf(`query {
-			result: applications(filter: {key:"%s", query: "\"%s\""}) {
-					%s
-			}
-	}`, nameKey, appName, b.gqlFieldsProvider.Page(b.gqlFieldsProvider.ForApplication())))
+func removeDoubleQuotesFromJSONKeys(in string) string {
```

Comment:
this func is used only in tests, shouldn't we move it to there?

Reply:
Removed